snow/engine/avalanche/state/vertex.go

```diff
@@ -16,8 +17,16 @@ import (
 	"github.com/ava-labs/avalanchego/utils/wrappers"
 )
 
-// maxSize is the maximum allowed vertex size. It is necessary to deter DoS.
-const maxSize = 1 << 20
+const (
+	// maxSize is the maximum allowed vertex size. It is necessary to deter DoS.
+	maxSize = 1 << 20
+
+	// maxNumParents is the max number of parents a vertex may have
+	maxNumParents = 128
+
+	// maxTxsPerVtx is the max number of transactions a vertex may have
+	maxTxsPerVtx = 128
```

Comment:
Since we are adding these in as constants, we should make sure that when calling BuildVertex, these values are honored... To keep the invariant that a virtuous node never generates an invalid vertex.